Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -828,7 +828,8 @@ void ContainerNode::removeChildren()
     Ref protectedThis { *this };
     NodeVector removedChildren;
     removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren);
-
+    for (Ref<Node>& node : removedChildren)
+        this->document().m_childrenToBeDeleted.append(RefPtr { node.ptr() });
```

Comment:
I think `appendRange` might do what you want here

Reply:
Done.